src/main.ts

```diff
@@ -425,11 +425,17 @@ function tryGetCSharpDevKitExtensionExports(csharpLogObserver: CsharpLoggerObser
     );
 }
 
+let registered = false;
```

Comment:
nit: I think we want a comment that this flag is to ensure that `profferBrokeredServices` is only done once.